src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1122,240 +1122,60 @@ protected static void VerifyOperationTreeForTest<TSyntaxNode>(CSharpCompilation
 
-            /// <summary>Enumerates the elements of a <see cref=""Span{T}""/>.</summary>
-            public ref struct Enumerator
-            {
-                /// <summary>The span being enumerated.</summary>
-                private readonly ReadOnlySpan<T> _span;
-                /// <summary>The next index to yield.</summary>
-                private int _index;
-
-                /// <summary>Initialize the enumerator.</summary>
-                /// <param name=""span"">The span to enumerate.</param>
-                internal Enumerator(ReadOnlySpan<T> span)
-                {
-                    _span = span;
-                    _index = -1;
-                }
-
-                /// <summary>Advances the enumerator to the next element of the span.</summary>
-                public bool MoveNext()
-                {
-                    int index = _index + 1;
-                    if (index < _span.Length)
-                    {
-                        _index = index;
-                        return true;
-                    }
-
-                    return false;
-                }
-
-                /// <summary>Gets the element at the current position of the enumerator.</summary>
-                public ref readonly T Current
-                {
-                    get => ref _span[_index];
-                }
-            }
-
-            public static implicit operator ReadOnlySpan<T>(T[] array) => array == null ? default : new ReadOnlySpan<T>(array);
-
-            public static implicit operator ReadOnlySpan<T>(string stringValue) => string.IsNullOrEmpty(stringValue) ? default : new ReadOnlySpan<T>((T[])(object)stringValue.ToCharArray());
+            return CreateCompilation(
+                text,
+                references: new List<MetadataReference>() { reference.EmitToImageReference() },
+                options: options,
+                parseOptions: parseOptions);
         }
 
-        public readonly ref struct SpanLike<T>
+        protected static CSharpCompilation CreateCompilationWithIndexAndRange(string text, CSharpCompilationOptions options = null, CSharpParseOptions parseOptions = null)
         {
-            public readonly Span<T> field;
-        }
+            var reference = CreateCompilation(new[] { TestSources.Index, TestSources.Range }).VerifyDiagnostics();
 
-        public enum Color: sbyte
-        {
-            Red,
-            Green,
-            Blue
+            return CreateCompilation(
+                text,
+                references: new List<MetadataReference>() { reference.EmitToImageReference() },
+                options: options,
+                parseOptions: parseOptions);
         }
 
-        public static unsafe class Helpers
+        protected static CSharpCompilation CreateCompilationWithIndexAndRangeAndSpan(string text, CSharpCompilationOptions options = null, CSharpParseOptions parseOptions = null)
```

Comment:
>CreateCompilationWithIndexAndRangeAndSpan [](start = 43, length = 41)

Rather than creating `CreateCompilation...` overloads (where the other parameters are unlikely to satisfy all callers), we should consider creating `static readonly MetadataReference` declarations for each of the additional source files.

Reply:
I'd rather have a couple of lines here, than having to repeat array initialization in hundreds of tests?
The use of these methods is typical anyways. The parameters sent don't change much.

---
In reply to: [185258129](https://github.com/dotnet/roslyn/pull/26405#discussion_r185258129) [](ancestors = 185258129)